neqo-transport/src/send_stream.rs

```diff
@@ -895,6 +908,70 @@ mod tests {
         assert_eq!(res, (15, None));
     }
 
+    #[test]
+    fn tx_buffer_next_bytes_1() {
+        let mut txb = TxBuffer::new();
+
+        assert_eq!(txb.avail(), 65535);
+
+        assert_eq!(txb.send(&[1; 100_000]), TxBuffer::BUFFER_SIZE);
+        assert!(matches!(txb.next_bytes(TxMode::Normal),
+			 Some((0, x)) if x.len()==65535));
```

Comment:
If you made the data you send into a constant, then could you just do a buffer comparison?

```suggestion
const DATA: &[u8] = &[1;100_000];
assert_eq!(txb.send(DATA), TxBuffer::BUFFER_SIZE);
assert_eq!(txb.next_bytes(TxMode::Normal), &DATA[..TxBuffer::BUFFER_SIZE]);
```

Reply:
I don't think it's that simple because `next_bytes` returns an Option, but I'll take the spirit of this and verify that the returned buffers contain what we expect.